money/src/main/proto/spine/money/money.proto

```diff
@@ -0,0 +1,990 @@
+// But in our type we use enumeration instead of a string currency code to provide more information
+// about currencies for UI-related code.
+//
+
+// An amount of money in its currency type.
+message Money {
+    // The currency of the amount of money.
+    Currency currency = 1 [(required) = true];
+
+    // The whole units of the amount.
+    //
+    // For example, if `currency` is `EUR`, then 1 unit is one euro.
+    int64 units = 2;
+
+    // Number of nano (10^-9) units of the amount.
+    //
+    // The value must be between -999,999,999 and +999,999,999 inclusive.
+    // If `units` is positive, `nanos` must be positive or zero.
+    // If `units` is zero, `nanos` can be positive, zero, or negative.
+    // If `units` is negative, `nanos` must be negative or zero.
+    // For example $-1.75 is represented as `units`=-1 and `nanos`=-750,000,000.
+    int32 nanos = 3;
+}
+
+// The ISO 4217 currency attributes (unless specifically noted).
+message CurrencyOptions {
+    // The name of the currency, e.g. "US dollar".
+    string name = 1;
+
+    // The symbol of the currency, e.g. "$".
+    // NOTE: ISO 4217 does not define any currency symbols.
+    string symbol = 2;
+
+    // The numeric code of the currency.
+    int32 numeric_code = 3;
+
+    // The number of digits after the decimal separator.
+    // The value is less than zero if there are no minor units for this currency (e.g., for gold).
+    // See ISO 4217 "exponent" definition for details.
+    int32 exponent_digits = 4;
+}
+
+extend google.protobuf.EnumValueOptions {
+    // 1025 is globally unique field number obtained from Google. Do not change it.
+    CurrencyOptions currency = 1025;
+}
+
+// The set of ISO 4217 currency alphabetic codes with other currency options (such as
+// currency name, numeric code etc).
+// NOTE: numeric codes must consist of 3 digits when displayed (e.g., "008", "050" etc).
+enum Currency {
+    CURRENCY_UNDEFINED = 0;
+    AED = 1 [(currency) = {
+        name: "UAE Dirham",
+        symbol: "د.إ",
+        numeric_code: 784,
+        exponent_digits: 2 }];
+    AFN = 2 [(currency) = {
+        name: "Afghani",
+        symbol: "؋",
+        numeric_code: 971,
+        exponent_digits: 2 }];
+    ALL = 3 [(currency) = {
+        name: "Lek",
+        symbol: "Lek",
+        numeric_code: 8,
+        exponent_digits: 2 }];
+    AMD = 4 [(currency) = {
+        name: "Armenian Dram",
+        symbol: "֏",
+        numeric_code: 51,
+        exponent_digits: 2 }];
+    ANG = 5 [(currency) = {
+        name: "Netherlands Antillean Guilder",
+        symbol: "ƒ",
+        numeric_code: 532,
+        exponent_digits: 2 }];
+    AOA = 6 [(currency) = {
+        name: "Kwanza",
+        symbol: "Kz",
+        numeric_code: 973,
+		exponent_digits: 2 }];
+    ARS = 7 [(currency) = {
+        name: "Argentine Peso",
+        symbol: "$",
+        numeric_code: 32,
+		exponent_digits: 2 }];
```

Comment:
Some weird formatting here and there. I think the idea was to indent `numeric_code` and `exponent_digits` additionally to improve readability. But that's not done for all enumeration members.